core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -55,10 +58,14 @@ private[spark] object ClosureCleaner extends Logging {
   private def getOuterClasses(obj: AnyRef): List[Class[_]] = {
```

Comment:
Should we merge this and `getOuterObjects` into one function that just returns pairs of objects and their classes? Maybe not just to keep the changes minimal?